api/kv.go

```diff
@@ -283,7 +286,36 @@ func (k *KV) deleteInternal(key string, params map[string]string, q *WriteOption
 
 // Txn is used to apply multiple KV operations in a single, atomic transaction.
 // Note that Go will perform the required base64 encoding on the values
-// automatically because the type is a byte slice.
+// automatically because the type is a byte slice. Transactions are defined as a
+// list of operations to perform, using the KVOp constants and KVTxnOp structure
+// to define operations. If any operation fails, none of the changes are applied
+// to the state store.
```

Comment:
What is the locking protocol used?  Suppose txn A updates keys 1 and 2, and transaction B updates keys 3, 2, and 1.  When executing the transaction, txn B updates 3, updates 2, and the thread is put to sleep.  Txn A comes through and completes its updates to keys 1 and 2.  Will the API replay the txn or will it outright fail because there was a concurrent update that caused txn A?

Said differently, can we wrap `AtomicApply()` with a helper function called `AtomicApplyDeadline()` (or `Retry` or `Timeout`) that will retry up to the {deadline,timeout,retry count}?  Where possible we should make it convenient for developers to not have to bother themselves with the consequences of concurrent and conflicting operations.  This gets sticky and why we should punt this control to the users w/ a new function interface.